File: c++_solutions/3101.cpp
class Solution {
public:
    long long countAlternatingSubarrays(vector<int>& nums) {
        long long arr_cnt = 0 ;
        int idx = 0 ;
        int prev_num = nums[0] ;
        int l = 0 , r = 0;
        while (idx < nums.size()){
            long long arr_len = 1;
            int curr_num = nums[idx] ; 
            idx++;
            while (idx < nums.size() && curr_num != nums[idx]){
                arr_len++;
                curr_num = nums[idx] ; 
                idx++;
            }
            arr_cnt += (arr_len * (arr_len+1))/2 ; 
        }
        return arr_cnt ; 
    }
};

File: python_solutions/3101.py
class Solution:
    def countAlternatingSubarrays(self, nums: List[int]) -> int:
        arr_cnt = 0 
        idx = 0 
        while idx < len(nums):
            curr_num = nums[idx] 
            arr_len = 1
            idx += 1
            while idx < len(nums) and nums[idx] != curr_num :
                curr_num = nums[idx]
                arr_len += 1
                idx += 1
            arr_cnt += (arr_len * (arr_len+1))//2
        return arr_cnt
        
